plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/loops/routing/OrthogonalSelfLoopRouter.java

```diff
@@ -0,0 +1,394 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.elk.alg.layered.intermediate.loops.routing;
+
+import org.eclipse.elk.alg.layered.graph.LEdge;
+import org.eclipse.elk.alg.layered.graph.LMargin;
+import org.eclipse.elk.alg.layered.graph.LNode;
+import org.eclipse.elk.alg.layered.graph.LPort;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfHyperLoop;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfHyperLoopLabels;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfLoopEdge;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfLoopHolder;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfLoopPort;
+import org.eclipse.elk.core.math.KVector;
+import org.eclipse.elk.core.math.KVectorChain;
+import org.eclipse.elk.core.options.PortSide;
+
+/**
+ * Routes self loops orthogonally. The routing functionality is exposed to subclasses. If a subclass wants to base its
+ * routing on the orthogonal routing, it can simply override {@link #modifyBendPoints(SelfLoopEdge, KVectorChain)} to
+ * turn an orthogonal routing into some kind of a strange, exotic, exciting new routing style!
+ */
+public class OrthogonalSelfLoopRouter extends AbstractSelfLoopRouter {
+    
+    /**
+     * The direction in which an edge goes around a node in its quest to reach its target.
+     */
+    protected enum EdgeRoutingDirection {
+        /** Clockwise direction. */
+        CLOCKWISE,
+        /** Counter-clockwise direction. */
+        COUNTER_CLOCKWISE;
+    }
+    
+    // TODO Replace by spacing options.
+    private static final double DISTANCE = 10.0;
+    public static final double LABEL_EDGE_DISTANCE = 5.0;
+
+    @Override
+    public void routeSelfLoops(final SelfLoopHolder slHolder) {
+        KVector nodeSize = slHolder.getLNode().getSize();
+        LMargin nodeMargins = slHolder.getLNode().getMargin();
+        
+        LMargin newNodeMargins = new LMargin();
+        newNodeMargins.set(nodeMargins);
+        
+        // Compute how far away from the node each routing slot on each side is
+        double[][] routingSlotPositions = computeRoutingSlotPositions(slHolder);
+        
+        for (SelfHyperLoop slLoop : slHolder.getSLHyperLoops()) {
+            for (SelfLoopEdge slEdge : slLoop.getSLEdges()) {
+                LEdge lEdge = slEdge.getLEdge();
+                
+                EdgeRoutingDirection routingDirection = computeEdgeRoutingDirection(slEdge);
+                
+                KVectorChain bendPoints = computeOrthogonalBendPoints(slEdge, routingDirection, routingSlotPositions);
+                bendPoints = modifyBendPoints(slEdge, routingDirection, bendPoints);
```

Comment:
Can you add a comment, in particular, that by default nothing happens here?

Reply:
Done.